azkaban-common/src/main/java/azkaban/project/JdbcProjectImpl.java

```diff
@@ -1015,31 +1017,37 @@ public void uploadFlowFile(final int projectId, final int projectVersion, final
 
   @Override
   public File getUploadedFlowFile(final int projectId, final int projectVersion,
-      final int flowVersion, final String flowName) throws ProjectManagerException {
+      final String flowFileName,
+      final int flowVersion) throws ProjectManagerException, IOException {
     final FlowFileResultHandler handler = new FlowFileResultHandler();
 
     final List<byte[]> data;
-    // Todo jamiesjc: delete the flow file after used.
-    final File file = new File(this.tempDir, flowName);
+    final File tempDir = Utils.createTempDir(this.tempDir);
```

Comment:
it's azkaban utils? I'm inclined towards using public OSS/JDK libs providing the method of the same purpose. They are more trustworthy and well tested.

Reply:
we also need to cover case where temp dir fails to be created for any reason. In this case might just throw an IOexception?

Reply:
Agreed. Changed to public jdk libs. 
Added the exception handling for failure to create temp dir.
Thanks!